### PR TITLE
docs/benchmarks: add benchmark result for 2.2

### PR DESCRIPTION
**Pull request by yichengq**
It also includes the script to run the benchmark.

for #3290

### CODE REVIEW

File: Documentation/benchmarks/etcd-2-2-benchmarks.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+| 64                | 256               | leader only        | 4204 | 73.7 |
+| 256               | 1                 | leader only        | 61 | 18.6 |
+| 256               | 64                | leader only        | 2431 | 32.0 |
+| 256               | 256               | leader only        | 4293 | 68.9 |
+| 64                | 64                | all servers        | 1766 | 54.3 |
+| 64                | 256               | all servers        | 4931 | 77.1 |
+| 256               | 64                | all servers        | 1878 | 50.9 |
+| 256               | 256               | all servers        | 4957 | 76.1 |
+
+## Caveat
```
Comment (xiang90) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: can we remove this part?

File: Documentation/benchmarks/bench.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash -e
```
Comment (xiang90) [inline Documentation/benchmarks/bench.sh]: we:
1.  separate this from the doc pr
2.  move this into its own dir
3.  have a readme about the script
4.  add etcd binary setup configuration into the script

File: Documentation/benchmarks/etcd-2-2-benchmarks.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+## Physical machines
```
Comment (xiang90) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: As I mentioned before, please compare the result with 2.1 and tell people why there is a change very briefly.

**Comment by yichengq** [comment]
updated as you suggest. PTAL

File: Documentation/benchmarks/etcd-2-2-benchmarks.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+Bootstrap another machine and use benchmark tool [boom](https://github.com/rakyll/boom) to send requests to each etcd member. Check [here](../../hack/benchmark/) for instructions.
+
+## Performance
+
+### reading one single key
+
+| key size in bytes | number of clients | target etcd server | read QPS | 90th Percentile Latency (ms) | read QPS (base) | 90th Percentile Latency (ms) (base) |
+|-------------------|-------------------|--------------------|----------|---------------|----|----|
+| 64                | 1                 | leader only        | 2216  | 0.5  | 2104  | 0.6 |
```
Comment (xiang90) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: i would expect to see +x% or -%x. just showing the exact pervious number is not clear enough.

File: Documentation/benchmarks/etcd-2-2-benchmarks.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+| 64                | 256               | leader only        | 2407 | 71.0 | 4206 | 69.3 |
+| 256               | 1                 | leader only        | 60   | 18.5 | 52   | 29.7 |
+| 256               | 64                | leader only        | 2186 | 37.2 | 1640 | 44.0 |
+| 256               | 256               | leader only        | 2385 | 81.9 | 4052 | 75.8 |
+| 64                | 64                | all servers        | 1758 | 53.1 | 1022 | 104.4 |
+| 64                | 256               | all servers        | 4547 | 86.7 | 3446 | 123.9 |
+| 256               | 64                | all servers        | 1667 | 54.7 | 1003 | 107.5 |
+| 256               | 256               | all servers        | 4695 | 81.3 | 3517 | 108.3 |
+
+### caveat
```
Comment (xiang90) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: caveat -> performance changes explanation

File: hack/benchmark/bench.sh [hunk 6]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash -e
+
+leader=http://10.240.201.15:4001
```
Comment (xiang90) [inline hack/benchmark/bench.sh]: ideally we should accept a flag. but this is fine for now.

File: Documentation/benchmarks/etcd-2-2-benchmarks.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+GCE n1-highcpu-2 machine type
+
+- 1x dedicated local SSD mounted under /var/lib/etcd
+- 1x dedicated slow disk for the OS
+- 1.8 GB memory
+- 2x CPUs
+
+## etcd Cluster
+
+3 etcd 2.2.0 members, each runs on a single machine.
```
Comment (xiang90) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: mention it is 2.2.0-alpha release

**Comment by xiang90** [comment]
LGTM. Thanks.

File: hack/benchmark/bench.sh (hunk 6 above)
Comment (yichengq) [inline hack/benchmark/bench.sh]: agree

File: Documentation/benchmarks/etcd-2-2-benchmarks.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+| 64                | 1                 | leader only        | 2216 (+5%) | 0.5 (-17%) | 2104  | 0.6 |
+| 64                | 64                | leader only        | 16038 (-10%) | 6.1 (+0%) | 17828 | 6.1 |
+| 64                | 256               | leader only        | 15497 (-16%) | 22.4 (+5%) | 18551 | 21.3 |
+| 256               | 1                 | leader only        | 2115  (-8%) | 0.5 (+0%) | 2291  | 0.5 |
+| 256               | 64                | leader only        | 16083 (-13%) | 6.1 (+8%) | 18407 | 5.6 |
+| 256               | 256               | leader only        | 15444 (-17%) | 21.9 (+2%) | 18403 | 21.3 |
+| 64                | 64                | all servers        | 45101 (-9%) | 2.1 (+5%) | 49089 | 2.0 |
+| 64                | 256               | all servers        | 50558 (-14%) | 8.0 (+8%) | 58129 | 7.4 |
+| 256               | 64                | all servers        | 45415 (-8%) | 2.1 (+5%) | 49207 | 2.0 |
+| 256               | 256               | all servers        | 50531 (-14%) | 8.1 (+20%) | 58725 | 6.7 |
```
Comment (yichengq) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: the `read QPS` looks quite narrow, and markdown doesn't support to adjust the width of column. :(

Comment (xiang90) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: can we remove the base? it does not provide much information. the precent is enough.

Comment (yichengq) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: let me try

Comment (yichengq) [inline Documentation/benchmarks/etcd-2-2-benchmarks.md]: much better now. :)

**Comment by xiang90** [comment]
lgtm again

**Pull request merged**